src/patterns.ts

```diff
@@ -0,0 +1,49 @@
+// TODO import directly from the grammar file?
```

Comment:
Was there a reason you didn’t start by importing from the grammar file?

Reply:
we could import directly from the grammar file if we changed the baseUrl in tsconfig from `src` to `.`

but I wasn't sure if that would break anything

Reply:
Yeah I think it may need tweaking and testing. It should work